vertical-pod-autoscaler/pkg/admission-controller/logic/server_test.go

```diff
@@ -286,3 +255,131 @@ func TestGetPatchesForResourceRequest(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateVPA(t *testing.T) {
+	badUpdateMode := vpa_types.UpdateMode("bad")
+	validUpdateMode := vpa_types.UpdateModeOff
+	badPolicyMode := vpa_types.ContainerScalingMode("bad")
+	validScalingMode := vpa_types.ContainerScalingModeAuto
+	tests := []struct {
+		name        string
+		vpa         vpa_types.VerticalPodAutoscaler
+		isCreate    bool
+		expectError error
+	}{
+		{
+			name: "empty ok",
+			vpa:  vpa_types.VerticalPodAutoscaler{},
```

Comment:
An empty create is wrong so is this an empty update?

Reply:
Done.